integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py

```diff
@@ -0,0 +1,155 @@
+
+
+class MongoDBAtlasDocumentStore:
+    def __init__(
+        self,
+        *,
+        mongo_connection_string: str,
+        database_name: str,
+        collection_name: str,
+        recreate_index: bool = False,
+    ):
+        """
+        Creates a new MongoDBAtlasDocumentStore instance.
+
+        This Document Store uses MongoDB Atlas as a backend (https://www.mongodb.com/docs/atlas/getting-started/).
+
+        :param mongo_connection_string: MongoDB Atlas connection string in the format:
+            "mongodb+srv://{mongo_atlas_username}:{mongo_atlas_password}@{mongo_atlas_host}/?{mongo_atlas_params_string}".
+            This can be obtained on the MongoDB Atlas Dashboard by clicking on the `CONNECT` button.
+        :param database_name: Name of the database to use.
+        :param collection_name: Name of the collection to use.
+        :param recreate_index: Whether to recreate the index when initializing the document store.
+        """
+        if collection_name and not bool(re.match(r"^[a-zA-Z0-9\-_]+$", collection_name)):
+            msg = f'Invalid collection name: "{collection_name}". It can only contain letters, numbers, -, or _.'
+            raise ValueError(msg)
+
+        self.mongo_connection_string = mongo_connection_string
+        self.database_name = database_name
+        self.collection_name = collection_name
+        self.recreate_index = recreate_index
+
+        self.connection: MongoClient = MongoClient(
+            self.mongo_connection_string, driver=DriverInfo(name="MongoDBAtlasHaystackIntegration")
+        )
+        self.database = self.connection[self.database_name]
+
+        if self.recreate_index:
+            self._get_collection().drop()
+
+        # Implicitly create the collection if it doesn't exist
+        if collection_name not in self.database.list_collection_names():
+            self.database.create_collection(self.collection_name)
+            self._get_collection().create_index("id", unique=True)
+
+    def _get_collection(self) -> Collection:
+        """
+        Returns the collection named by index or returns the collection specified when the
+        driver was initialized.
+        """
+        return self.database[self.collection_name]
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Utility function that serializes this Document Store's configuration into a dictionary.
+        """
+        return default_to_dict(
+            self,
+            mongo_connection_string=self.mongo_connection_string,
+            database_name=self.database_name,
+            collection_name=self.collection_name,
+            recreate_index=self.recreate_index,
+        )
+
+    def count_documents(self, filters: Optional[Dict[str, Any]] = None) -> int:
+        """
+        Returns how many documents are present in the document store.
+        """
+        return self._get_collection().count_documents({} if filters is None else filters)
```

Comment:
Nitpick - Let's extract the argument into its own variable. Will make it easier to debug.